src/main/java/teammates/common/datatransfer/attributes/AdminEmailAttributes.java

```diff
@@ -15,37 +16,49 @@
 import teammates.storage.entity.AdminEmail;
 
 public class AdminEmailAttributes extends EntityAttributes<AdminEmail> {
-
-    public String emailId;
+    // Required fields
     public List<String> addressReceiver;
     public List<String> groupReceiver;
     public String subject;
+    public Text content;
+
+    // Optional fields
     public Date sendDate;
     public Date createDate;
-    public Text content;
+    public String emailId;
     public boolean isInTrashBin;
 
-    public AdminEmailAttributes(AdminEmail ae) {
-        this.emailId = ae.getEmailId();
-        this.addressReceiver = ae.getAddressReceiver();
-        this.groupReceiver = ae.getGroupReceiver();
-        this.subject = ae.getSubject();
-        this.sendDate = ae.getSendDate();
-        this.createDate = ae.getCreateDate();
-        this.content = ae.getContent();
-        this.isInTrashBin = ae.getIsInTrashBin();
-    }
-
-    public AdminEmailAttributes(String subject,
-                                List<String> addressReceiver,
-                                List<String> groupReceiver,
-                                Text content,
-                                Date sendDate) {
-        this.subject = subject;
-        this.addressReceiver = addressReceiver;
-        this.groupReceiver = groupReceiver;
-        this.content = content;
-        this.sendDate = sendDate;
+    AdminEmailAttributes() {
+        createDate = Const.TIME_REPRESENTS_DEFAULT_TIMESTAMP;
+        emailId = Const.ParamsNames.ADMIN_EMAIL_ID;
+        isInTrashBin = false;
+    }
+
+    /**
+     * Creates a new AdminEmail with default values for optional fields.
+     *
+     * <p>Following default values are set to corresponding attributes:
+     * <ul>
+     * <li>{@code false} for {@code isInTrashBin}</li>
+     * <li>{@code Const.TIME_REPRESENTS_DEFAULT_TIMESTAMP} for {@code createDate}</li>
+     * <li>{@code Const.ParamsNames.ADMIN_EMAIL_ID} for {@code emailId}</li>
+     * </ul>
+     */
+    public static Builder builder(String subject, List<String> addressReceiver, List<String> groupReceiver, Text content) {
+        return new Builder(subject, addressReceiver, groupReceiver, content);
+    }
+
+    public static AdminEmailAttributes valueOf(AdminEmail adminEmail) {
+        return new Builder(
+                adminEmail.getSubject(),
+                adminEmail.getAddressReceiver(),
+                adminEmail.getGroupReceiver(),
+                adminEmail.getContent())
```

Comment:
Looks not really easy to read. This might be better:
```java
return new Builder(adminEmail.getSubject(), adminEmail.getAddressReceiver(),
                adminEmail.getGroupReceiver(), adminEmail.getContent())
        .with*(*)
        .build();
```